jdaviz/core/template_mixin.py

```diff
@@ -1314,13 +1318,78 @@ def add_results_from_plugin(self, data_item):
             self.app.data_collection.remove(self.app.data_collection[self.label])
 
         self.app.add_data(data_item, self.label)
+        if ref_data is not None:
+            self.link_results_from_plugin(ref_data)
 
         if self.add_to_viewer_selected != 'None':
             # replace the contents in the selected viewer with the results from this plugin
             # TODO: switch to an instance/classname check?
             self.app.add_data_to_viewer(self.viewer.selected_id,
                                         self.label, clear_other_data=replace)
 
+    def link_results_from_plugin(self, ref_data):
+        dc = self.app.data_collection
+        ref_data = dc[ref_data.selected_item['label']]
+        plugin_data = dc[self.label]
+
+        len_ref_pixel = len(ref_data.pixel_component_ids)
+        len_plugin_pixel = len(plugin_data.pixel_component_ids)
+        if ref_data.coords is not None and plugin_data.coords is not None:
+            # Link by WCS if possible
+            dc.add_link(WCSLink(ref_data, plugin_data))
+
+        elif len_ref_pixel > 0 and len_plugin_pixel > 0:
+            # If WCS does not exist in one of the data, use pixel linking instead
+            # pc_ref = _get_pixel_coordinate_axes(ref_data)
+            # pc_plugin = _get_pixel_coordinate_axes(plugin_data)
+            pc_ref = [str(id).split(" ")[-1][1] for id in ref_data.pixel_component_ids]
+            pc_plugin = [str(id).split(" ")[-1][1] for id in plugin_data.pixel_component_ids]
+
+            links = []
+
+            # This code loops through the returned locations of the x, y, and z
+            # axes in the pixel_coordinate_ids of the reference data. It matches
+            # the axes with the pixel_coordinate_ids of the plugin data and links
+            # those axes. There are special rules for linking 2D and 3D data, which
+            # is as follows: 2D y to 3D z and 2D x to 3D y, and vice versa in the
+            # case of moment map and collapse data because they need to be transposed.
+            # See github comment:
+            # https://github.com/spacetelescope/jdaviz/pull/1412#discussion_r907630682
+            for ind, pixel_coord in enumerate(pc_ref):
+                ref_index = ind
+                if (len_plugin_pixel == 2 and
+                        (plugin_data.meta.get("Plugin", None) == 'MomentMap' or
+                         plugin_data.meta.get("Plugin", None) == 'Collapse')):
```

Comment:
will we need more exceptions for future plugins (is there any other way to generalize this check than relying on the plugin names themselves)?

```suggestion
                        (plugin_data.meta.get("Plugin", None) in ('MomentMap', 'Collapse')):
```

Reply:
from offline discussions: looks like GaussianSmooth should also be considered in this list as that technically results in a cube being returned which is then collapsed into the profile viewer.

Reply:
thinking about this more... are there any known cases that would not apply if we just rely on the `len_plugin_pixel == 2` check alone (so link for all plugin outputs that have 2 pixel axes rather than listing individual plugins)?